pkg/handler/handler_test.go

```diff
@@ -52,22 +52,27 @@ func TestRegisterOpenAPIVersionedService(t *testing.T) {
 	client := server.Client()
 
 	tcs := []struct {
-		acceptHeader string
-		respStatus   int
-		respBody     []byte
+		acceptHeader              string
+		respStatus                int
+		responseContentTypeHeader string
```

Comment:
I would also recommend checking that `mime.ParseMediaType(resp.Header.Get("Content-Type"))` succeeds on the returned content type, so we have an in-repo test that the response will work with the response to all three requested content-types (same comment in the handler3 test)

Reply:
Love the idea!